pinot-core/src/test/java/org/apache/pinot/queries/MultiValueRawQueriesTest.java

```diff
@@ -468,6 +907,80 @@ private void validateSimpleAggregateQueryResults(ResultTable resultTable, DataSc
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.LONG, DataSchema.ColumnDataType.LONG, DataSchema.ColumnDataType.DOUBLE,
+          DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE,
+          DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE,
+          DataSchema.ColumnDataType.DOUBLE
+      });
+      validateAggregateWithFilterQueryResults(resultTable, dataSchema);
+    }
+    {
+      // Aggregation on double columns with filter
+      String query = "SELECT COUNTMV(mvDoubleCol), COUNTMV(mvRawDoubleCol), SUMMV(mvDoubleCol), SUMMV(mvRawDoubleCol), "
+          + "MINMV(mvDoubleCol), MINMV(mvRawDoubleCol), MAXMV(mvDoubleCol), MAXMV(mvRawDoubleCol), AVGMV(mvDoubleCol), "
+          + "AVGMV(mvRawDoubleCol) from testTable WHERE mvRawDoubleCol > 1000.0";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+
+      DataSchema dataSchema = new DataSchema(new String[]{
+          "countmv(mvDoubleCol)", "countmv(mvRawDoubleCol)", "summv(mvDoubleCol)", "summv(mvRawDoubleCol)",
+          "minmv(mvDoubleCol)", "minmv(mvRawDoubleCol)", "maxmv(mvDoubleCol)", "maxmv(mvRawDoubleCol)",
+          "avgmv(mvDoubleCol)", "avgmv(mvRawDoubleCol)"
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.LONG, DataSchema.ColumnDataType.LONG, DataSchema.ColumnDataType.DOUBLE,
+          DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE,
+          DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.DOUBLE,
+          DataSchema.ColumnDataType.DOUBLE
+      });
+      validateAggregateWithFilterQueryResults(resultTable, dataSchema);
+    }
+  }
+
+  private void validateAggregateWithFilterQueryResults(ResultTable resultTable, DataSchema expectedDataSchema) {
+    assertNotNull(resultTable);
+    assertEquals(resultTable.getDataSchema(), expectedDataSchema);
+    List<Object[]> recordRows = resultTable.getRows();
+    assertEquals(recordRows.size(), 1);
+
+    Object[] values = recordRows.get(0);
+    long count = (long) values[0];
+    long countRaw = (long) values[1];
+    assertEquals(count, 80);
+    assertEquals(count, countRaw);
+
+    double sum = (double) values[2];
+    double sumRaw = (double) values[3];
+    assertEquals(sum, 84360.0);
+    assertEquals(sum, sumRaw);
+
+    double min = (double) values[4];
+    double minRaw = (double) values[5];
+    assertEquals(min, 1000.0);
+    assertEquals(min, minRaw);
+
+    double max = (double) values[6];
+    double maxRaw = (double) values[7];
+    assertEquals(max, 1109.0);
+    assertEquals(max, maxRaw);
+
+    double avg = (double) values[8];
+    double avgRaw = (double) values[9];
+    assertEquals(avg, 1054.5);
+    assertEquals(avg, avgRaw);
+  }
+
   @Test
   public void testAggregateWithGroupByQueries() {
```

Comment:
Have we covered following scenarios ?

- single GROUP BY MV column (raw)
- 2 or more GROUP BY MV columns (all raw)
- 2 or more GROUP BY MV columns (mix of raw and dict)
- 2 or more GROUP BY MV columns (all raw) along 1 or more GROUP BY SV column (dict)
- 2 or more GROUP BY MV columns (mix of raw and dict) along 1 or more GROUP BY SV column (raw or dict)

Reply:
Also, can we add transform (on the MV raw group by column) to one of the aggregate group by queries ?